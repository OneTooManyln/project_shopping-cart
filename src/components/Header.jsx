export default function Header() {
  return (
    <>
      <header className="flex flex-col bg-[rgb(251,113,153)] h-50 w-full">
        <div className="w-screen h-6"></div>
        <div className="header-banner flex justify-between bg-white border-y-0 border-y-black max-lg:justify-center">
          <img
            src="/src/assets/Hangzhou_Spark_wordmark.svg"
            alt="Page Header Logo"
            className="object-contain h-20 mt-5 mb-1"
          />
          <div className="box-wrapper flex flex-1 justify-end mr-20 gap-5 max-lg:hidden">
            <div id="box" className="w-20"></div>
            <div id="box" className="w-20"></div>
            <div id="box" className="w-20"></div>
            <div id="box" className="w-20"></div>
            <div id="box" className="w-20"></div>
          </div>
        </div>
        <div className="banner-bottom flex justify-center w-screen /*h-[10px]*/ g-[rgb(87,136,206)] bg-white border-b-[1px] border-[rgb(251,113,153)] mt-[3px] py-[10px]"></div>
      </header>
    </>
  );
}
