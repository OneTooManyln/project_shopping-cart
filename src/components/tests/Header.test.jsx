import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Header from "../Header";

describe("Header component", () => {
  it("renders without errors", () => {
    render(<Header />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders logo", () => {
    render(<Header />);
    expect(screen.getByAltText(/Page Header Logo/i)).toBeInTheDocument();
  });
});
