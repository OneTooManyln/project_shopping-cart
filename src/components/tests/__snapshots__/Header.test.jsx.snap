// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex flex-col bg-[rgb(251,113,153)] h-50 w-full"
  >
    <div
      class="w-screen h-6"
    />
    <div
      class="header-banner flex justify-between bg-white border-y-0 border-y-black"
    >
      <img
        alt="Page Header Logo"
        class="object-contain h-20 mt-5 mb-1"
        src="/src/assets/Hangzhou_Spark_wordmark.svg"
      />
      <div
        class="box-wrapper flex flex-1 justify-end mr-20 gap-5"
      >
        <div
          class="w-20"
          id="box"
        />
        <div
          class="w-20"
          id="box"
        />
        <div
          class="w-20"
          id="box"
        />
        <div
          class="w-20"
          id="box"
        />
        <div
          class="w-20"
          id="box"
        />
      </div>
    </div>
    <div
      class="banner-bottom flex justify-center w-screen /*h-[10px]*/ g-[rgb(87,136,206)] bg-white border-b-[1px] border-[rgb(251,113,153)] mt-[3px] py-[10px]"
    />
  </header>
</DocumentFragment>
`;
